perf_test/sparse/KokkosSparse_block_pcg.cpp

```diff
@@ -0,0 +1,743 @@
+      delete [] adj;
+      delete [] ew;
+
+
+      values_view_t kok_x_original = create_x_vector<values_view_t>(((nv /block_size) + 1) * block_size, MAXVAL);
+      for (INDEX_TYPE i = nv; i < ((nv /block_size) + 1) * block_size; ++i){
+    	  kok_x_original(i) = 0;
+      }
+      run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original, block_size);
+
+      myExecSpace::finalize();
+    }
+
+#endif
+
+#if defined( KOKKOS_ENABLE_CUDA )
+    if ( cmdline[ CMD_USE_CUDA ] ) {
+      // Use the last device:
+      INDEX_TYPE nv = 0, ne = 0;
+      INDEX_TYPE *xadj, *adj;
+      SCALAR_TYPE *ew;
+      Kokkos::HostSpace::execution_space::initialize();
+      Kokkos::Cuda::initialize( Kokkos::Cuda::SelectDevice( cmdline[ CMD_USE_CUDA_DEV ] ) );
+      Kokkos::Cuda::print_configuration(std::cout);
+
+      KokkosKernels::Impl::read_matrix<INDEX_TYPE,INDEX_TYPE, SCALAR_TYPE> (&nv, &ne, &xadj, &adj, &ew, mtx_bin_file);
+
+
+      typedef Kokkos::Cuda myExecSpace;
+      typedef typename KokkosSparse::CrsMatrix<SCALAR_TYPE, INDEX_TYPE, myExecSpace, void, SIZE_TYPE > crsMat_t;
+
+      typedef typename crsMat_t::StaticCrsGraphType graph_t;
+      typedef typename crsMat_t::row_map_type::non_const_type row_map_view_t;
+      typedef typename crsMat_t::index_type::non_const_type   cols_view_t;
+      typedef typename crsMat_t::values_type::non_const_type values_view_t;
+
+      row_map_view_t rowmap_view("rowmap_view", nv+1);
+      cols_view_t columns_view("colsmap_view", ne);
+      values_view_t values_view("values_view", ne);
+
+
+      {
+        typename row_map_view_t::HostMirror hr = Kokkos::create_mirror_view (rowmap_view);
+        typename cols_view_t::HostMirror hc = Kokkos::create_mirror_view (columns_view);
+        typename values_view_t::HostMirror hv = Kokkos::create_mirror_view (values_view);
+
+        for (INDEX_TYPE i = 0; i <= nv; ++i){
+          hr(i) = xadj[i];
+        }
+
+        for (INDEX_TYPE i = 0; i < ne; ++i){
+          hc(i) = adj[i];
+          hv(i) = ew[i];
+        }
+        Kokkos::deep_copy (rowmap_view , hr);
+        Kokkos::deep_copy (columns_view , hc);
+        Kokkos::deep_copy (values_view , hv);
+
+
+      }
+      graph_t static_graph (columns_view, rowmap_view);
+      crsMat_t crsmat("CrsMatrix", nv, values_view, static_graph);
+
+//      typedef typename KokkosSparse::CrsMatrix<SCALAR_TYPE, INDEX_TYPE, Kokkos::Cuda> crsMat_t;
+//      crsMat_t crsmat("CrsMatrix", nv, nv, ne, ew, xadj, adj);
+      delete [] xadj;
+      delete [] adj;
+      delete [] ew;
+
+      values_view_t kok_x_original = create_x_vector<values_view_t>(((nv /block_size) + 1) * block_size, MAXVAL);
+/*
+      for (INDEX_TYPE i = nv; i < ((nv /block_size) + 1) * block_size; ++i){
+    	  kok_x_original(i) = 0;
+      }
+*/
+      run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original, block_size);
+
+      //run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original);
+
+      myExecSpace::finalize();
```

Comment:
I think calling finalize() for a specific execution space is to be deprecated in Kokkos, @crtrott.